packages/eslint-plugin/docs/rules/no-wrapper-object-types.mdx

```diff
@@ -0,0 +1,74 @@
+---
+description: 'Disallow using confusing built-in primitive class wrappers.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-wrapper-object-types** for documentation.
+
+The JavaScript language includes a set of primitives such as `boolean` and `number`s.
+JavaScript also contains a "wrapper" class object for each of those primitives, such as [`Boolean`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean) and [`Number`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number).
+TypeScript represents the primitive in lower-case types and instances of the wrapper classes with upper-case.
+
+Using the primitives like `0` instead of class constructors like `Number(0)` is generally considered a JavaScript best practice.
+As a result, using the lower-case primitive type names like `number` instead of class names like `Number` is generally considered a TypeScript best practice.
+
+Similarly, TypeScript's `Function` type refers to objects created with the `Function` constructor, i.e. `new Function("eval'd code with implicit args")`.
```

Comment:
I think this part implies that uppercase types and wrapper constructor calls are equivalent, which is technically imprecise. The `Function` type denotes anything that _looks like_ a function (i.e. structurally equivalent to its lib interface). This includes classes, etc., which you can't create with the `Function` constructor. Similarly, `number` is a subtype of `Number` basically.

Reply:
Hmm yeah I see what you mean. But I'm having a hard time thinking of better phrasing 😬 do you have any suggestions @Josh-Cena?